src/sdl/index.ts

```diff
@@ -105,6 +107,28 @@ export class SDL {
     }
   }
 
+  static validateStorage(name: string, storage: v2ResourceStorage | v2ResourceStorageArray | undefined) {
+    if (!storage) {
+      throw new Error("Storage is required for service " + name);
```

Comment:
this looks a bit weird. If it's required it should be required on type level. Or optional in type

Reply:
Perhaps we can add a schema to validate this in `YAML.load(yaml)`, but how it works; users input their SDL/yaml and we want to validate that the required properties are properly populated.

Reply:
So perhaps it makes sence that these fields are optional in the type itself? This way whoever works with the code realistically knows what to expect

Reply:
They're required, but you can send whatever yaml file in there.

Reply:
Then it should be unknown and resolve with a valid a typed sdl. Would this make sense?

Reply:
Makes sense